### ISSUE TITLE
[a11y]: Tag - Playground - Screen Reader is not announcing the Disable Status for the Tag

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#16746]
### Package

@carbon/react

### Browser

Firefox

### Package version

v1.58.1

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

Once we set Disable Boolean to "True" in Playground Screen Reader is not announcing the Disable status for the Tag.

### Recommended Fix
Ensure jaws should announce as the unavailable for the Tag content when it is disabled.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tag--playground&args=disabled:!true

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
